cycletls/roundtripper.go

```diff
@@ -155,6 +125,38 @@ func (rt *roundTripper) dialTLS(ctx context.Context, network, addr string) (net.
 	return nil, errProtocolNegotiated
 }
 
+func (rt *roundTripper) DoDialTlsContext(ctx context.Context, network, addr string) (conn *utls.UConn, err error) {
+
+	var host string
+	if host, _, err = net.SplitHostPort(addr); err != nil {
+		host = addr
+	}
+
+	rawConn, err := rt.dialer.DialContext(ctx, network, addr)
+	if err != nil {
+		return nil, err
+	}
+
+	var spec *utls.ClientHelloSpec
+	if len(rt.JA3) > 0 {
+		spec, err = StringToSpec(rt.JA3, rt.UserAgent, rt.forceHTTP1)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
we should maybe raise an error if the JA3 isn't passed in or default it, otherwise it doesn't make a lot of sense to have the `if len(rt.JA3) > 0 {` check